### PR TITLE
CSHARP-3745: Add support for srvServiceName URI option

### PR DESCRIPTION
**Pull request by adelinowona**
- Added a new URI options `srvServiceName` which allows the user to change the srv address prefix from the default of `mongodb`

### CODE REVIEW

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Review by BorisDog — CHANGES_REQUESTED** [review]

File: src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs [hunk 1]
```diff
@@ -928,6 +937,11 @@ private void Parse()
                 throw new MongoConfigurationException("Specifying srvMaxHosts when connecting to a replica set is invalid.");
             }
 
+            if (!_isInternalRepresentation && _srvServiceName != null && _scheme != ConnectionStringScheme.MongoDBPlusSrv)
```
[the remark this answers is not in the document]
Comment (adelinowona) [reply src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs:940]: So `_isInternalRepresentation` helps to differentiate between a connection string provided by the user and one that has been transformed into a `mongodb://` after the SRV and TXT records are resolved from DNS. So we need to check the connection string before the transformation has happened.

File: src/MongoDB.Driver/MongoDefaults.cs [hunk 2]
```diff
@@ -223,6 +224,14 @@ public static TimeSpan SocketTimeout
             set { __socketTimeout = value; }
         }
 
+        /// <summary>
+        /// Gets the SRV service name.
+        /// </summary>
+        public static string SrvServiceName
```
[the remark this answers is not in the document]
Comment (adelinowona) [reply src/MongoDB.Driver/MongoDefaults.cs]: Done!

File: src/MongoDB.Driver/MongoUrlBuilder.cs [hunk 3]
```diff
@@ -1183,6 +1199,7 @@ private void InitializeFromConnectionString(ConnectionString connectionString)
             _serverSelectionTimeout = connectionString.ServerSelectionTimeout.GetValueOrDefault(MongoDefaults.ServerSelectionTimeout);
             _socketTimeout = connectionString.SocketTimeout.GetValueOrDefault(MongoDefaults.SocketTimeout);
             _srvMaxHosts = connectionString.SrvMaxHosts;
+            _srvServiceName = connectionString.SrvServiceName;
```
[the remark this answers is not in the document]
Comment (adelinowona) [reply src/MongoDB.Driver/MongoUrlBuilder.cs]: Done

File: tests/MongoDB.Driver.Core.Tests/Core/Configuration/ClusterSettingsTests.cs [hunk 4]
```diff
@@ -316,6 +317,15 @@ public void Constructor_with_negative_srvMaxHosts_should_throw()
...
         }
 
+        [Fact]
+        public void Constructor_with_srvServiceName_should_initialize_instance()
+        {
+            var srvServiceName = "customname";
+            var subject = new ClusterSettings(srvServiceName: srvServiceName);
+
+            subject.SrvServiceName.Should().Be(srvServiceName);
+        }
```
[the remark this answers is not in the document]
Comment (adelinowona) [reply tests/MongoDB.Driver.Core.Tests/Core/Configuration/ClusterSettingsTests.cs:327]: Done

File: src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs [hunk 5]
```diff
@@ -485,6 +488,12 @@ public ConnectionStringScheme Scheme
         /// </summary>
         public int? SrvMaxHosts => _srvMaxHosts;
 
+        /// <summary>
+        /// Gets the SRV service name which modifies the srv URI to look like:
+        /// <code>_{srvServiceName}._tcp.{hostname}.{domainname}</code>
+        /// </summary>
+        public string SrvServiceName => _srvServiceName;
```
[the remark this answers is not in the document]
Comment (adelinowona) [reply src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs:495]: Done

File: tests/MongoDB.Driver.Tests/MongoClientSettingsTests.cs [hunk 6]
```diff
@@ -677,6 +677,12 @@ public void TestFreezeInvalid()
                 settings.SrvMaxHosts = 2;
             });
 
+            AssertException(settings =>
```
[the remark this answers is not in the document]
Comment (adelinowona) [reply tests/MongoDB.Driver.Tests/MongoClientSettingsTests.cs:680]: Done

**Review by BorisDog — APPROVED** [review]
LGTM

**Pull request merged**